### PR TITLE
New Extension: SpeciesType

### PR DESCRIPTION
**Pull request by DavidSagan**
This pull request has a new file for standardizing the names of particle species.

*Implements issue:* #176

## Description

This convention is for standardizing the names of particle species, e.g. in particle physics.

## Affected Components

- new extension: `SpeciesType`

## Logic Changes

A new *optional* string attribute `speciesType` can now be present at `mesh record` and `particle group` level.

## Writer Changes

- [ ] `openPMD-api` (upcoming): Add to writer and reader API.

## Reader Changes

Readers can if they want read the new attribute (if it exists) and propose e.g. groups, accumulations, etc. on it.

- [x] `openPMD-validator`: Validate it's a string if present https://github.com/openPMD/openPMD-validator/pull/38
- [ ] `openPMD-viewer`: https://github.com/openPMD/openPMD-viewer/...
- [ ] `yt`: https://github.com/yt-project/yt/...
- [ ] `VisIt`: https://github.com/openPMD/openPMD-visit-plugin
- [ ] `openPMD-api` (upcoming): https://github.com/openPMD/openPMD-api/...

## Data Updater

New optional extension with optional attribute. No update of existing data needed.
Edit: update needed for `ED-PIC` extension: #192

### CODE REVIEW

**Comment by ax3l** [comment]
Left-over to do's from #177:

- [x] hardons: define protons & anti-protons
- [x] separate anti-particles by `,` instead of putting them in brackets
- [x] remove extension-ID (2.0.0)

**Comment by DavidSagan** [comment]
@ax3l There also should be a version number associated with this. I assume it should be 1.0.0 .

**Comment by ax3l** [comment]
@DavidSagan the version number will simply be the one of the standard it gets in with.

**Review by ax3l — APPROVED** [review]

**Comment by DavidSagan** [comment]
@ax3l 

> the version number will simply be the one of the standard it gets in with.

OK just to make sure I understand: Say in the future changes to an extension necessitates that it is to be bumped up a version. Then even if the base standard has not changed it will be bumped up too? Yes?

**Comment by ax3l** [comment]
Exactly.
Admittedly, up to now the major changes were triggered by the base standard.

**Comment by ax3l** [comment]
@RemiLehe @marcguetg hi, do you want to give your review comments on this PR? From my side I think it's already in quite good shape.

File: EXT_SpeciesType.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+------------
+
+This convention is for standardizing the names of particle species, e.g. in
+particle physics.
+
+
+Additional Record Attribute
+---------------------------
+
+The following additional attribute for openPMD `records` is defined in this
```
Comment (ax3l) [inline EXT_SpeciesType.md]: oh wait, I think it should read:

... *mesh `records`* and *particle* `groups` ...

Comment (DavidSagan) [reply EXT_SpeciesType.md]: How about simply:

The following attribute for openPMD is defined ...

Comment (ax3l) [reply EXT_SpeciesType.md]: Sure, but I think for reader tools it is good to know where to look for it, no?
I currently only see these two locations as useful and relaxing later on is always possible.

Comment (DavidSagan) [reply EXT_SpeciesType.md]: OK I did not realize that this was defining where the info should go. How about adding a definition for `mesh record` and `particle group` in the Standard?

**Review by RemiLehe — APPROVED** [review]

**Comment by RemiLehe** [comment]
@ax3l Should we merge this?

File: EXT_SpeciesType.md (hunk 1 above)
Comment (ax3l) [reply EXT_SpeciesType.md]: good, this aspect is moved to #187

**Comment by ax3l** [comment]
ok, I addressed the last inline comment about the location of the `SpeciesType` attribute, so reader implementors know where to expect it

**Comment by DavidSagan** [comment]
Minor detail: Since `neutron` is a common standard name do we want to use `neutron`, `anti-neutron` rather than `other:neutron`, `other:anti-neutron`?

**Comment by ax3l** [comment]
Sure, we can do that. It's just not too common for us since it's neutral ;-)

I did just draw a line (arbitrarily) since (anti-)protons are the only stable free hadrons.

**Comment by ax3l** [comment]
@DavidSagan moved (anti-)neutron to "common" and added sigma baryon as another example. now we draw the line at the nucleus :D

File: EXT_SpeciesType.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+particle physics.
+
+
+Additional Record Attribute
+---------------------------
+
+The following additional attribute for openPMD `mesh records` and
+`particle groups` is defined in this extension:
+
+- `SpeciesType`
```
Comment (ax3l) [inline EXT_SpeciesType.md]: wait, why is this upper-case in the first letter? Aren't we starting all attributes with a lowercase letter? Will update for consistency.

**Review by ax3l — CHANGES_REQUESTED** [review]

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
@RemiLehe @DavidSagan I think this can be merged now :)

File: EXT_SpeciesType.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  - `other:kaon`, ...
+
+### Atoms & Isotopes
+
+Element namings follow the abbreviated namings of the periodic table, defined
+by *The International Union of Pure and Applied Chemistry* (IUPAC).
+An example would be `Si` for silicon.
+
+Specifications of isotopes are denoted by a pound symbol `#` followed
+by the isotopic number followed by the chemical symbol, e.g.: `#3He`
```
Comment (zerothi) [inline EXT_SpeciesType.md:87]: Could we add a mention on how to mention `#2H2#15O`? It isn't fully clear how this works in this flow?

**Pull request merged**